pkg/networkaware/networkoverhead/networkoverhead.go

```diff
@@ -0,0 +1,577 @@
+		}
+	}
+	klog.V(4).Infof("Scores after normalization: %s", scores)
+	return nil
+}
+
+// MinMax : get min and max scores from NodeScoreList
+func getMinMaxScores(scores framework.NodeScoreList) (int64, int64) {
+	var max int64 = math.MinInt64 // Set to min value
+	var min int64 = math.MaxInt64 // Set to max value
+
+	for _, nodeScore := range scores {
+		if nodeScore.Score > max {
+			max = nodeScore.Score
+		}
+		if nodeScore.Score < min {
+			min = nodeScore.Score
+		}
+	}
+
+	return min, max
+}
+
+// sortNetworkTopologyCosts : sort costs if manual weights were selected
+func sortNetworkTopologyCosts(no *NetworkOverhead, networkTopology *ntv1alpha1.NetworkTopology) {
+	if no.weightsName != ntv1alpha1.NetworkTopologyNetperfCosts { // Manual weights were selected
+		for _, w := range networkTopology.Spec.Weights {
+			// Sort Costs by TopologyKey, might not be sorted since were manually defined
+			sort.Sort(networkawareutil.ByTopologyKey(w.TopologyList))
+		}
+	}
+	return
+}
+
+// populateCostMap : Populates costMap based on the node being filtered/scored
+func populateCostMap(no *NetworkOverhead, costMap map[networkawareutil.CostKey]int64, networkTopology *ntv1alpha1.NetworkTopology, region string, zone string) {
+	seen := false
+	for _, w := range networkTopology.Spec.Weights { // Check the weights List
+		if w.Name == no.weightsName { // If its the Preferred algorithm
+			if region != "" { // Add Region Costs
+				// Binary search through CostList: find the Topology Key for region
+				topologyList := networkawareutil.FindTopologyKey(w.TopologyList, ntv1alpha1.NetworkTopologyRegion)
+
+				if no.weightsName != ntv1alpha1.NetworkTopologyNetperfCosts {
+					// Sort Costs by origin, might not be sorted since were manually defined
+					sort.Sort(networkawareutil.ByOrigin(topologyList))
+				}
+
+				// Binary search through TopologyList: find the costs for the given Region
+				costs := networkawareutil.FindOriginCosts(topologyList, region)
+
+				// Add Region Costs
+				for _, c := range costs {
+					costMap[networkawareutil.CostKey{ // Add the cost to the map
+						Origin:      region,
+						Destination: c.Destination}] = c.NetworkCost
+				}
+			}
+			if zone != "" { // Add Zone Costs
+				// Binary search through CostList: find the Topology Key for zone
+				topologyList := networkawareutil.FindTopologyKey(w.TopologyList, ntv1alpha1.NetworkTopologyZone)
+
+				if no.weightsName != ntv1alpha1.NetworkTopologyNetperfCosts {
+					// Sort Costs by origin, might not be sorted since were manually defined
+					sort.Sort(networkawareutil.ByOrigin(topologyList))
+				}
+
+				// Binary search through TopologyList: find the costs for the given Region
+				costs := networkawareutil.FindOriginCosts(topologyList, zone)
+
+				// Add Zone Costs
+				for _, c := range costs {
+					costMap[networkawareutil.CostKey{ // Add the cost to the map
+						Origin:      zone,
+						Destination: c.Destination}] = c.NetworkCost
+				}
+			}
+			seen = true
```

Comment:
Isn't it equivalent to an immediate `return` from this entire function?

Reply:
Yes. Code updated.